weave/flow/dataset.py

```diff
@@ -71,6 +75,64 @@ def to_pandas(self) -> "pd.DataFrame":
 
         return pd.DataFrame(self.rows)
 
+    def add_rows(self, rows: Iterable[dict]) -> "Dataset":
+        """Create a new dataset version by appending rows to the existing dataset.
+
+        This is useful for adding examples to large datasets without having to
+        load the entire dataset into memory.
+
+        Args:
+            rows: The rows to add to the dataset.
+
+        Returns:
+            The updated dataset.
+        """
+        client = get_weave_client()
+        if client is None:
+            raise ValueError("Weave is not initialized, call weave.init().")
+        if not isinstance(self.rows, WeaveTable) or not self.rows.table_ref:
+            raise TypeError(
+                "This dataset is not saved to weave. Call weave.publish(dataset) "
+                "to save the dataset before adding rows."
+            )
+        if self.rows.table_ref.project != client.project:
+            raise ValueError(
+                "This dataset is not saved to the same project as the current weave client. "
+                "Client is in project:  "
+                + client.project
+                + " but dataset is in project: "
+                + self.rows.table_ref.project
+            )
+        if self.rows.table_ref.entity != client.entity:
+            raise ValueError(
+                "This dataset is not saved to the same entity as the current weave client. "
+                "Client is in entity: "
+                + client.entity
+                + " but dataset is in entity: "
+                + self.rows.table_ref.entity
+            )
+        table_update_res = client._append_to_table(
+            self.rows.table_ref.digest, list(rows)
+        )
+        new_table_ref = TableRef(
+            entity=client.entity,
+            project=client.project,
+            _digest=table_update_res.digest,
+        ).uri()
+        return client._save_object(
+            {
+                # str() to prevent weave from capturing values as refs
```

Comment:
I don't understand, what does this mean... example?

Reply:
<img width="234" alt="Screenshot 2025-04-24 at 10 13 56 AM" src="https://github.com/user-attachments/assets/8e6c0455-c728-4ba0-9986-7c412b449dfb" />

The name field and description fields will be shown in the oject versions table as an attr reference, instead of the plain string value